pkg/model/resource/options.go

```diff
@@ -0,0 +1,259 @@
+	// error as flags may seem to be missing because Cobra assigned them to another flag.
+	if strings.HasPrefix(opts.Group, "-") {
+		return fmt.Errorf(groupRequired)
+	}
+	if strings.HasPrefix(opts.Version, "-") {
+		return fmt.Errorf(versionRequired)
+	}
+	if strings.HasPrefix(opts.Kind, "-") {
+		return fmt.Errorf(kindRequired)
+	}
+	// Now we can check that all the required flags are not empty
+	if len(opts.Group) == 0 {
+		return fmt.Errorf(groupRequired)
+	}
+	if len(opts.Version) == 0 {
+		return fmt.Errorf(versionRequired)
+	}
+	if len(opts.Kind) == 0 {
+		return fmt.Errorf(kindRequired)
+	}
+
+	// Check if the Group has a valid DNS1123 subdomain value
+	if err := IsDNS1123Subdomain(opts.Group); err != nil {
+		return fmt.Errorf("group name is invalid: (%v)", err)
+	}
+
+	// Check if the version follows the valid pattern
+	if !versionRegex.MatchString(opts.Version) {
+		return fmt.Errorf(
+			"version must match %s (was %s)", versionPattern, opts.Version)
+	}
+
+	// Check if the Kind is PascalCase
+	if opts.Kind != flect.Pascalize(opts.Kind) {
+		return fmt.Errorf("kind must be PascalCase (expected %s was %s)", flect.Pascalize(opts.Kind), opts.Kind)
+	}
+
+	// TODO: validate plural strings if provided
+
+	return nil
+}
+
+func (opts *Options) GVK() config.GVK {
+	return config.GVK{
+		Group:   opts.Group,
+		Version: opts.Version,
+		Kind:    opts.Kind,
+	}
+}
+
+// safeImport returns a cleaned version of the provided string that can be used for imports
+func (opts *Options) safeImport(unsafe string) string {
+	safe := unsafe
+
+	// Remove dashes and dots
+	safe = strings.Replace(safe, "-", "", -1)
+	safe = strings.Replace(safe, ".", "", -1)
+
+	return safe
+}
+
+func (opts *Options) NewV1Resource(c *config.Config, doResource bool) *Resource {
+	res := opts.newResource()
+
+	// NOTE: while directories can have "-" and ".", v1 needs that directory to be a Go package
+	//       and that is the reason why we remove them for the directory
+	pkg := path.Join(c.Repo, "pkg", "apis", res.GroupPackageName, opts.Version)
+	domain := c.Domain
+
+	// pkg and domain may need to be changed in case we are referring to a builtin core resource:
+	//  - Check if we are scaffolding the resource now           => project resource
+	//  - Check if we already scaffolded the resource            => project resource
+	//  - Check if the resource group is a well-known core group => builtin core resource
+	//  - In any other case, default to                          => project resource
+	// TODO: need to support '--resource-pkg-path' flag for specifying resourcePath
+	if !doResource {
+		file := filepath.Join("pkg", "apis", res.GroupPackageName, opts.Version,
+			fmt.Sprintf("%s_types.go", strings.ToLower(opts.Kind)))
+		if _, err := os.Stat(file); os.IsNotExist(err) {
+			if coreDomain, found := v1CoreGroups[opts.Group]; found {
+				pkg = path.Join("k8s.io", "api", opts.Group, opts.Version)
+				domain = coreDomain
+			}
+		}
+	}
```

Comment:
> Before your refactor, for v1, we'd do the check and build the universe with `policy` treated as core.
> 
> Then, we'd _completely_ ignore that, generate the API anyway, and then _re-run that check in a different method_ when generating the controller. The second time we run the check, API exists, so the file exists, so we treat policy as local.

@DirectXMan12 what do you think about this logic for v1? Basically it enforces it to be a project-local resource if `doResource` flag is set (after asking the user in case he didnt specify it) and defaults to looking for the file instead.